src/mesh/p4est_mesh.jl

```diff
@@ -7,8 +7,8 @@ to manage trees and mesh refinement.
 !!! warning "Experimental code"
     This mesh type is experimental and can change any time.
 """
-mutable struct P4estMesh{NDIMS, RealT<:Real, NDIMSP2, NNODES} <: AbstractMesh{NDIMS}
-  p4est                 ::Ptr{p4est_t}
+mutable struct P4estMesh{NDIMS, RealT<:Real, P, NDIMSP2, NNODES} <: AbstractMesh{NDIMS}
+  p4est                 ::P # Either Ptr{p4est_t} or Ptr{p8est_t}
```

Comment:
`P == `Ptr{p4est_t}` if `NDIMS == 2` and `P == `Ptr{p8est_t}` if `NDIMS == 3`, correct? Would it make sense to add a check of the correct type of `p4est` to the inner constructor?